go/libraries/doltcore/schema/tag.go

```diff
@@ -0,0 +1,72 @@
+// Copyright 2020 Liquidata, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package schema
+
+import (
+	"crypto/sha512"
+	"encoding/binary"
+	"math/rand"
+
+	"github.com/liquidata-inc/dolt/go/store/types"
+)
+
+const (
+	// TODO: increase ReservedTagMin to 1 << 63 once numeric marshalling is fixed
+	// ReservedTagMin is the start of a range of tags which the user should not be able to use in their schemas.
+	ReservedTagMin uint64 = 1 << 50
```

Comment:
We're now limited to 1,125,899,906,842,624 columns in a repo. You've been warned.